src/components/article/EditArticle.jsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { withRouter, Redirect } from 'react-router-dom';
+import ArticleForm from './ArticleForm';
+import DisplayMessage from '../presentation/DisplayMessage';
+import { getArticle, editArticle } from '../../actions/articles';
+import PreLoader from '../presentation/PreLoader';
+
+
+class EditArticle extends React.Component {
+  state = {
+    title: '',
+    body: '',
+    tags: [],
+    imgUrl: '',
+    message: false,
+    isFetching: true,
+  };
+
+  async componentDidMount() {
+    const { article, getArticle: getArticleToEdit, match: { params: { slug } } } = this.props;
+    await getArticleToEdit(slug);
+    await this.setState({
+      title: article.title,
+      body: article.body,
+      tags: article.tags,
+      isFetching: false,
+    });
+  }
+
+  onSubmit = async (formState) => {
+    try {
+      const { editArticle: updateArticle, history, match: { params: { slug } } } = this.props;
+
+      await this.setState({
+        title: formState.title,
+        body: formState.body,
+        tags: formState.tags,
+        imgUrl: formState.imgUrl,
+      });
+      await updateArticle(slug, this.state);
+      const articleslug = `/articles/${slug}`;
+      return history.push(articleslug);
+    } catch (error) {
+      return null;
+    }
+  };
+
+  render() {
+    const {
+      message,
+      isFetching,
+    } = this.state;
+    const { isLoggedIn, article } = this.props;
+    const articleTags = article.tags ? article.tags.join() : '';
+    if (isLoggedIn === null) return null;
+    if (!isLoggedIn) return <Redirect to="/login" />;
```

Comment:
There'd be no need for this once you make this route a private one like I pointed out above